packages/react-i18n/docs/react_i18n_schema.md

```diff
@@ -0,0 +1,295 @@
+
+If a node has child leaf KVPs for each of the pluralization keys required by the language, then its children are within a pluralization context.
+
+Any additional children within the context are invalid.
+
+# Serialization
+
+React I18n can be serialized to different formats. However, not all serialization formats are created equal.
+
+Two serializations are officialy defined. Consumers of React I18n `1.0` files are expected to be able to parse files serialized into each of these serialization formats.
+
+| Serialization Format | JSON+Comments | JSON |
+| -------------------- | ------------- | ---- |
+| Comments             | ✅             | ❌    |
+
+## Serialization to JSON
+
+When serialized to [JSON](https://www.json.org/json-en.html), each of the dictionaries become JSON `object` types, and the keys and leaf values are both JSON `string` types.
+
+**Example React I18n JSON file**
+
+```json
+{
+  "hello": "My name is {name}",
+  "parent": {
+    "child_1": "I am the first child of `parent`!",
+    "child_2": {
+      "grandchild_1": "I am the first grandchild of `parent`!",
+      "grandchild_2": "I am the second grandchild of `parent`!"
+    }
+  }
+}
+```
+
+### Comments
+
+React I18n does not support the use of comments when serialized to JSON. This is due to JSON's lack of support for comments.
+
+## Serialization to JSON+Comments
+
+`JSON+Comments` is a serialization format whose syntax is a superset of `JSON`, but a subset of [`JSON5`](https://json5.org/).
+
+```
+JSON < JSON+Comments < JSON5
+```
+
+Historically, `JSON` had support for comments, but this [was then removed](https://archive.vn/20150704102718/https://plus.google.com/+DouglasCrockfordEsq/posts/RK8qyGVaGSr). However, comments have proven to be a useful mechanism for developers to [provide additional context to translators](https://development.shopify.io/engineering/developing_at_Shopify/internationalization/providing_context_notes). The `JSON+Comments` serialization format extends the `JSON` serialization to re-add support for comments.
+
+The `JSON+Comments` can be parsed using a `JSON5` parser, or a `JSON` parser that is not sensitive to comments.
+
+**Example React I18n JSON+Commments file**
+
+```json5
+{
+  // This is a context comment for `hello`!
+  "hello": "My name is {name}",
+  "parent": {
+    "child_1": "I am the first child of `parent`!", // This is a trailing context comment for `child_1`!
+    "child_2": {
+      /*
+        This is a multi-line
+        context comment for `grandchild_1`!
+      */
+      "grandchild_1": "I am the first grandchild of `parent`!",
+      "grandchild_2": "I am the second grandchild of `parent`!" /* This is a trailing context comment for `grandchild_2`! */
+    }
+  }
+}
+```
+
+Comments follow the [JSON5 syntax for comments](https://spec.json5.org/#comments): `//` signals the start of a comment that ends at the end of the line. `/*` signals the start of a comment that ends with the next `*/`.
```

Comment:
This tangent about JSON5 and comments is a bit irrelevant to the schema definition don't you think? Can we move to another doc that talks about serialization for those interested and link it here?

Reply:
Removed.